File: count_orders.c
#include "zipcode_data.h"
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
  load_file(argv[1]);
  int zipcode;
  int count;
  for (int i = 2; i < argc; i++) {
    zipcode = strtol(argv[i], (char **)NULL, 10);
    count = count_orders(zipcode);
    printf("%i\n", count);
  }
  return 0;
}

File: count_orders.py
import sys
import argparse

data = []


def count_orders(zipcode):
    global data
    return data.count(zipcode)


def load_file(name):
    global data
    with open(name, "r") as f:
        data = [int(x) for x in f.readlines()]


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("datafile")
    parser.add_argument("zipcodes", type=int, nargs="*")
    args = parser.parse_args(sys.argv[1:])
    load_file(args.datafile)
    for zipcode in args.zipcodes:
        print(count_orders(zipcode))


if __name__ == "__main__":
    main()
